src/components/Modal.svelte

```diff
@@ -1,53 +1,41 @@
 <script lang="ts">
-  export let floating = false;
-  export let error = false;
-  export let subtle = false;
-  export let small = false;
-  export let narrow = false;
-  export let center = false;
+  import type { ComponentProps } from "svelte";
+
+  import { twemoji } from "@app/lib/utils";
+  import * as modal from "@app/lib/modal";
+
+  import Button from "@app/components/Button.svelte";
+
+  export let error: boolean = false;
+  export let emoji: string | undefined = error ? "🚨" : undefined;
+  export let title: string | undefined = undefined;
+  export let body: string | undefined = undefined;
+
+  export let showCloseAction: boolean = true;
+  export let closeActionName: string = "Close";
+  export let closeActionCallback: () => void = modal.hide;
+  export let closeActionProps: Partial<ComponentProps<Button>> = {};
+
+  export let primaryAction:
+    | {
+        name: string;
+        callback: () => void;
+        props?: Partial<ComponentProps<Button>>;
+      }
+    | undefined = undefined;
```

Comment:
Hmm I would suggest to encode the `closeAction` equal to the `primaryAction` something like
EDIT: if we want to not declare some props in the `closeAction` object, we can spread some defaults into the prop

```suggestion
  interface Action {
    name: string;
    callback: () => void;
    props?: Partial<ComponentProps<Button>>;
  } 
  
  export let closeAction: Partial<Action>;
  closeAction = {...closeAction,  ...{
    name: "Close",
    callback: modal.hide,
    props: {}
  }};
  
  export let primaryAction: Action | undefined = undefined;
```

Reply:
How about 0c6283d4c0a8ae511940dd0965fd270df62de587?

Reply:
yeah that seems nice 👍